src/storage/key.rs

```diff
@@ -0,0 +1,94 @@
+use std::fmt::Debug;
+use std::cmp::Ordering;
+
+/// Trait `Key`
+pub trait Key<'a>:
+    AsRef<[u8]> + From<Vec<u8>> + Debug + Clone + Send + Sync + Ord + Default
+{
+    /// Key must have fixed length
+    const LEN: u16;
+
+    /// Reference type for zero-copy key creation
+    type Ref: RefKey<'a>;
+
+    /// Convert `Self` into `Vec<u8>`
+    fn to_vec(&self) -> Vec<u8> {
+        self.as_ref().to_vec()
+    }
+
+    /// Convert `Self` to `Self::Ref`
+    fn as_ref_key(&'a self) -> Self::Ref {
+        Self::Ref::from(self.as_ref())
+    }
+}
+
+/// Trait for reference key type
+pub trait RefKey<'a>: Ord + From<&'a [u8]> {}
+
+/// Key for demonstration purposes
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct VectorKey(Vec<u8>);
```

Comment:
Maybe we can implement `VectorKey` with constant generic parameter to specify size?
```
pub struct VectorKey<const N: usize>(Vec<u8>);
```